symphony-bdk-template/symphony-bdk-template-api/src/main/java/com/symphony/bdk/template/api/TemplateEngine.java

```diff
@@ -0,0 +1,72 @@
+package com.symphony.bdk.template.api;
+
+import org.apiguardian.api.API;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Interface to represent a specific template engine backed by a specific technology.
+ * It creates {@link Template} instances based on a file provided in the file system, in the classpath or as a URL
+ */
+@API(status = API.Status.STABLE)
+public interface TemplateEngine {
+
+  /**
+   * Returns the names of built-in templates which can be retrieved by calling {@link #newBuiltInTemplate(String)}
+   * @return the names of available built-in templates
+   */
+  Set<String> getBuiltInTemplates();
+
+  /**
+   * Creates a new {@link Template} from the built-in template name.
+   * @param template the name of one of the built-in templates returned by {@link #getBuiltInTemplates()}
+   * @return a new {@link Template} instantiated from the provided template name
+   * @throws TemplateException if template not found
+   */
+  Template newBuiltInTemplate(String template) throws TemplateException;
+
+  /**
+   * Create a {@link Template} instance from a file on the file system
+   * @param templatePath path to a template file on the file system
+   * @return a new {@link Template} instantiated from the provided file
+   * @throws TemplateException when template cannot be loaded, e.g. file not accessible
+   */
+  Template newTemplateFromFile(String templatePath) throws TemplateException;
```

Comment:
perhaps we could support URI? and then find out it is a file or the classpath based on it?

file:///root/blah.text classpath://com/symphony/bdk/blah.text